### PR TITLE
Various simplifications, bug fixes and removals of deprecated APIs

### PR DESCRIPTION
**Pull request by bvwells**
Various simplifications, bug fixes and removals of deprecated APIs.

### CODE REVIEW

**Comment by bvwells** [comment]
Hope this is useful!

File: prow/spyglass/lenses/lenses.go [hunk 1]
```diff
@@ -91,9 +91,6 @@ func RegisterLens(lens Lens) error {
 	if config.Title == "" {
 		return errors.New("empty title field in view metadata")
 	}
-	if config.Priority < 0 {
```
Comment (alvaroaleman) [inline prow/spyglass/lenses/lenses.go:94]: Why did you remove this?

File: config/tests/jobs/jobs_test.go [hunk 2]
```diff
@@ -753,11 +753,6 @@ func checkScenarioArgs(jobName, imageName string, args []string) error {
 	}
 
 	if scenario == "" {
-		entry := jobName
-		if strings.HasPrefix(jobName, "pull-security-kubernetes") {
```
Comment (alvaroaleman) [inline config/tests/jobs/jobs_test.go:757]: Why did you remove this? Do we not ready the `entry` variable anymore?

File: prow/kube/client.go [hunk 3]
```diff
@@ -214,8 +214,6 @@ func (c *Client) requestRetry(r *request) ([]byte, error) {
 		return nil, NewConflictError(fmt.Errorf("body: %s", string(rb)))
 	} else if resp.StatusCode == 422 {
 		return nil, NewUnprocessableEntityError(fmt.Errorf("body: %s", string(rb)))
-	} else if resp.StatusCode == 404 {
```
Comment (alvaroaleman) [inline prow/kube/client.go:217]: Why did you remove this?

File: prow/repoowners/repoowners.go [hunk 4]
```diff
@@ -794,7 +794,7 @@ func (o *RepoOwners) entriesForFile(path string, people map[string]map[*regexp.R
 	if !o.enableMDYAML || !strings.HasSuffix(path, ".md") {
 		// if path is a directory, this will remove the leaf directory, and returns "." for topmost dir
 		d = filepath.Dir(d)
-		d = canonicalize(path)
```
Comment (alvaroaleman) [inline prow/repoowners/repoowners.go:797]: This changes the result, are you sure this is correct?

**Review by alvaroaleman — COMMENTED** [review]
/ok-to-test
Thanks for your work! Some questions, but most of it looks good

File: prow/spyglass/lenses/lenses.go (hunk 1 above)
Comment (bvwells) [reply prow/spyglass/lenses/lenses.go:94]: Priority is a uint so can't be less than zero.

File: prow/kube/client.go (hunk 3 above)
Comment (bvwells) [reply prow/kube/client.go:217]: 404 is already handled just above. Sadly GitHub diff doesn't show it. :-)

File: prow/repoowners/repoowners.go (hunk 4 above)
Comment (bvwells) [reply prow/repoowners/repoowners.go:797]: Not 100%. The value for d from 
```
d = filepath.Dir(d)
```
Is never used as it is overwritten by
```
d = canonicalize(path)
```
And is hence redundant. There is some code in the same file which does 
```
                d = filepath.Dir(d)
		d = canonicalize(d)
```
So I assume this should be the same...

**Comment by bvwells** [comment]
Thanks for the review @alvaroaleman!

File: config/tests/jobs/jobs_test.go (hunk 2 above)
Comment (bvwells) [reply config/tests/jobs/jobs_test.go:757]: Yes, ```entry``` is no longer used.

File: prow/repoowners/repoowners.go (hunk 4 above)
Comment (alvaroaleman) [reply prow/repoowners/repoowners.go:797]: Can you instead change it to just `d = canonicalize(d)` (without the `filepath.Dir(d)`)? That would preserve the current behavior. Unfortunately we don't have tests for this it seems

Comment (bvwells) [reply prow/repoowners/repoowners.go:797]: Sure, no problems. I've updated and removed the comment as it refers to ```filepath.Dir(d)```.

**Review by alvaroaleman — APPROVED** [review]
/assign @fejta

File: kubetest/aksengine.go [hunk 5]
```diff
@@ -214,7 +214,11 @@ func (c *aksEngineDeployer) SetCustomCloudProfileEnvironment() error {
 			Timeout: 30 * time.Second,
 		}
 		endpointsresp, err := httpClient.Get(metadataURL)
-		if err != nil || endpointsresp.StatusCode != 200 {
+		if err != nil {
```
Comment (spiffxp) [inline kubetest/aksengine.go:217]: I don't think this matches the intent of what was here before.  Is there a test that demonstrates the behavior this is simplifying?

https://golang.org/pkg/net/http/#Client.Get

> A non-2xx response doesn't cause an error.

File: prow/flagutil/github.go [hunk 6]
```diff
@@ -57,9 +57,9 @@ func (o *GitHubOptions) AddFlags(fs *flag.FlagSet) {
 // Validate validates GitHub options.
 func (o *GitHubOptions) Validate(dryRun bool) error {
 	endpoints := o.endpoint.Strings()
-	for _, uri := range endpoints {
+	for i, uri := range endpoints {
 		if uri == "" {
-			uri = github.DefaultAPIEndpoint
+			endpoints[i] = github.DefaultAPIEndpoint
```
Comment (spiffxp) [inline prow/flagutil/github.go:62]: can you add a test that demonstrates the bug that this is fixing?

File: prow/github/helpers.go [hunk 7]
```diff
@@ -34,7 +34,7 @@ const ImageSizeLimit = 5242880
 // HasLabel checks if label is in the label set "issueLabels".
 func HasLabel(label string, issueLabels []Label) bool {
 	for _, l := range issueLabels {
-		if strings.ToLower(l.Name) == strings.ToLower(label) {
+		if strings.EqualFold(l.Name, label) {
```
Comment (spiffxp) [inline prow/github/helpers.go:37]: does this more closely match GitHub's equality than what was there before?

File: prow/github/types.go [hunk 8]
```diff
@@ -745,7 +745,7 @@ func (i Issue) IsPullRequest() bool {
 // HasLabel checks if an issue has a given label.
 func (i Issue) HasLabel(labelToFind string) bool {
 	for _, label := range i.Labels {
-		if strings.ToLower(label.Name) == strings.ToLower(labelToFind) {
+		if strings.EqualFold(label.Name, labelToFind) {
```
Comment (spiffxp) [inline prow/github/types.go:748]: does this more closely match GitHub's equality than what was there before?

File: prow/pluginhelp/hook/hook.go [hunk 9]
```diff
@@ -362,13 +362,13 @@ func (ha *HelpAgent) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "405 Method not allowed", http.StatusMethodNotAllowed)
 		return
 	}
-	b, err := ioutil.ReadAll(r.Body)
+	_, err := ioutil.ReadAll(r.Body)
```
Comment (spiffxp) [inline prow/pluginhelp/hook/hook.go:365]: Should we consider removing this block entirely? ref: https://github.com/kubernetes/test-infra/pull/15010

**Review by spiffxp — CHANGES_REQUESTED** [review]
A few questions/nits

Out of curiosity, what tool were you using for this?

File: prow/pluginhelp/hook/hook.go (hunk 9 above)
Comment (bvwells) [reply prow/pluginhelp/hook/hook.go:365]: Sure, removed.

File: prow/github/helpers.go (hunk 7 above)
Comment (bvwells) [reply prow/github/helpers.go:37]: https://golang.org/pkg/strings/#EqualFold
```
EqualFold reports whether s and t, interpreted as UTF-8 strings, are equal under Unicode case-folding, which is a more general form of case-insensitivity.
```

You would perhaps never hit any issues as the labels you are using are in English. Use of strings.EqualFold is often encouraged as a best practice.

File: prow/github/types.go (hunk 8 above)
Comment (bvwells) [reply prow/github/types.go:748]: See https://github.com/kubernetes/test-infra/pull/19444#discussion_r499798993

File: prow/flagutil/github.go (hunk 6 above)
Comment (bvwells) [reply prow/flagutil/github.go:62]: Added a number of tests for the Validate function. Cleaned up Validate slightly as some of the code there is unreachable. Hope it is all okay.

**Comment by bvwells** [comment]
@spiffxp thanks for the review. I was using https://github.com/golangci/golangci-lint which runs many community linters. 

Not sure if this is something you'd be interested in running on CI. I can raise an issue if it is of interest.

File: prow/flagutil/github.go (hunk 6 above)
Comment (bvwells) [reply prow/flagutil/github.go:62]: The bug here is that the range loop over endpoints takes a copy of each slice element. The copy of the slice element ```uri``` is set to the default API endpoint which has absolutely no effect.

I think that the expected behaviour here should be that if any of the string elements in the endpoints slice is empty its value should be defaulted to the default API endpoint.

**Comment by alvaroaleman** [comment]
/lgtm

ping @spiffxp anything open from your side?

btw @bvwells does this fix all erros golangci-lint finds (in which configuration?)? We could consider enabling it, as its pretty good at finding subtile bugs and dead code

**Comment by bvwells** [comment]
@alvaroaleman there are still more issues to be fixed. Some of them would depend on which linters you'd like to enable. I could raise an issue if you like.

**Review by fejta — APPROVED** [review]

**Comment by alvaroaleman** [comment]
> @alvaroaleman there are still more issues to be fixed. Some of them would depend on which linters you'd like to enable. I could raise an issue if you like.

Yeah, if you can do that, that would be great. Its a topic that's perfectly suited for newcomes and opening an issue will tell ppl that this is something we want.
